web/themes/joinup/joinup_theme.theme

```diff
@@ -101,28 +128,90 @@ function joinup_theme_preprocess_fieldset(&$variables) {
 }
 
 /**
- * Implements hook_theme_suggestions_alter().
+ * Implements hook_preprocess_HOOK() for container.html.twig.
  */
-function joinup_theme_theme_suggestions_alter(array &$suggestions, array $variables) {
-  if (isset($variables['element']) && isset($variables['element']['#type'])) {
-    if ($variables['element']['#type'] == 'form') {
-      $original_theme_hook = $variables['theme_hook_original'];
-      $suggestions[] = $original_theme_hook . '__' . str_replace('-', '_', $variables['element']['#id']);
-    }
-    if ($variables['theme_hook_original'] == 'form_element') {
-      $original_theme_hook = $variables['theme_hook_original'];
-      $suggestions[] = $original_theme_hook . '__' . str_replace('-', '_', $variables['element']['#type']);
-    }
+function joinup_theme_preprocess_container(&$variables) {
+  $route_match = \Drupal::routeMatch();
+  $route_name = $route_match->getRouteName();
+  $form_routes = array(
+    'node.add',
+    'rdf_entity.rdf_add',
+    'entity.node.edit_form',
+    'collection.propose_form',
+    'entity.rdf_entity.edit_form',
+  );
 
-    $slider_ids = array(
-      'edit-field-ar-elibrary-creation',
-      'edit-field-test-lista',
-      'edit-field-test-luist-normal',
-    );
+  // Check if container is field element - it has only one parent.
+  if (!empty($route_name) && in_array($route_name, $form_routes)
+    && isset($variables['element']['#parents'])
+    && count($variables['element']['#parents']) == 1
+    && $variables['element']['#id'] != 'edit-actions') {
+    $variables['is_form_wrapper'] = TRUE;
+  }
+}
 
-    if ($variables['theme_hook_original'] == 'select' && in_array($variables['element']['#id'], $slider_ids)) {
-      $original_theme_hook = $variables['theme_hook_original'];
-      $suggestions[] = $original_theme_hook . '__slider';
+/**
+ * Implements hook_theme_suggestions_HOOK_alter() for form-element.html.twig.
+ */
+function joinup_theme_theme_suggestions_form_element_alter(array &$suggestions, array $variables) {
+  $suggestions[] = 'form_element__' . str_replace('-', '_', $variables['element']['#type']);
+}
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter() for select.html.twig.
+ */
+function joinup_theme_theme_suggestions_select_alter(array &$suggestions, array $variables) {
+  // Array contains all select elements ids, which are displayed as slider.
+  $slider_ids = array(
+    'edit-field-ar-elibrary-creation',
+  );
+
+  if (in_array($variables['element']['#id'], $slider_ids)) {
+    $suggestions[] = 'select__slider';
+  }
+}
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter() for input.html.twig.
+ */
+function joinup_theme_theme_suggestions_input_alter(array &$suggestions, array $variables) {
+  if ($variables['element']['#type'] == 'submit') {
+    $button_value = Html::cleanCssIdentifier($variables['element']['#value'], array(' ' => '_'));
```

Comment:
Using the button value as theme suggestion is not really reliable, because in case of multilingual or label changes it's going to break. I don't see any templates committed for this suggestion, is it still needed?